cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -142,6 +153,46 @@ func (s *server) addArchiveStorage(opts *querysvc.QueryServiceOptions, host comp
 	return nil
 }
 
+func (s *server) addV2ArchiveStorage(opts *v2querysvc.QueryServiceOptions, host component.Host) error {
+	if s.config.Storage.TracesArchive == "" {
+		s.telset.Logger.Info("Archive storage not configured")
+		return nil
+	}
+
+	f, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesArchive, host)
+	if err != nil {
+		return fmt.Errorf("cannot find archive storage factory: %w", err)
+	}
+
+	archiveFactory, ok := f.(storage.ArchiveFactory)
+	if !ok {
+		s.telset.Logger.Info("Archive storage not supported by the factory")
+	}
+	reader, err := archiveFactory.CreateArchiveSpanReader()
+	if errors.Is(err, storage.ErrArchiveStorageNotConfigured) || errors.Is(err, storage.ErrArchiveStorageNotSupported) {
+		s.telset.Logger.Info("Archive storage not created", zap.String("reason", err.Error()))
+	}
+	if err != nil {
+		s.telset.Logger.Error("Cannot init archive storage reader", zap.Error(err))
+	}
+	writer, err := archiveFactory.CreateArchiveSpanWriter()
+	if errors.Is(err, storage.ErrArchiveStorageNotConfigured) || errors.Is(err, storage.ErrArchiveStorageNotSupported) {
+		s.telset.Logger.Info("Archive storage not created", zap.String("reason", err.Error()))
+	}
+	if err != nil {
+		s.telset.Logger.Error("Cannot init archive storage writer", zap.Error(err))
+	}
+
+	if writer != nil && reader != nil {
+		opts.ArchiveTraceReader = v1adapter.NewTraceReader(reader)
+		opts.ArchiveTraceWriter = v1adapter.NewTraceWriter(writer)
+	} else {
+		s.telset.Logger.Info("Archive storage not initialized")
+	}
```

Comment:
@yurishkuro we probably want to break this out into a helper since its being used in two palces. Do you have any thoughts for where this helper should live? I was initially thinking it should be a function on the `v2querysvc.QueryServiceOptions`. However, that would introduce a dependency on the v1 storage interface.

Reply:
ok to leave for now. We haven't finished archive factory PR, but the idea was that in the v2 extension we don't be using `v1adapter` but get v2 reader/writer directly from storage extension.